### PR TITLE
Check for old witness_node_data_dir

### LINKED ISSUE #2340: detect './witness_node_data_dir' for backwards-compat
With https://github.com/steemit/steem/issues/2176, the default data dir location changes; for a smoother experience for users who are upgrading from prior versions (and using defaults), steemd should first check for the old-style `./witness_node_data_dir`. Otherwise it may be confusing to have a new build resync even though there's an intact data dir nearby.

### PR DESCRIPTION
**Pull request by mvandeberg**
Closes #2340

### CODE REVIEW

**Comment by roadscape** [comment]
Unless I'm mistaken it looks like this prevents users from running `steemd --data-dir ./witness_node_data_dir`; this check should be avoided if it was explicitly set.

Also maybe the message could say:
```
Please move your data directory, or launch steemd with --data-dir ./witness_node_data_dir
```

**Comment by mvandeberg** [comment]
The logic is in an else block after `if( my->_args.count("data-dir")  )`, so `data-dir=./witness_node_data_dir` will still work.

I am going to wait on more feedback before changing the message (if we want to) to avoid unnecessary builds.

File: libraries/appbase/application.cpp [hunk 1]
```diff
@@ -136,6 +136,15 @@ bool application::initialize_impl(int argc, char** argv, vector<abstract_plugin*
          app_dir << '.' << app_name;
 
          data_dir = data_dir / app_dir.str();
+
+         #pragma message( "TODO: Remove this check for Steem release 0.20.1+" )
+         bfs::path old_dir = bfs::current_path() / "witness_node_data_dir";
+         if( bfs::exists( old_dir ) )
+         {
+            std::cerr << "The default data directory is now " << data_dir << " instead of " << old_dir << ".\n";
```
[the remark this answers is not in the document]
Comment (mvandeberg) [reply libraries/appbase/application.cpp]: No, if the user specifies a data dir, this code block is never executed.

https://github.com/steemit/steem/blob/2340-witness-node-data-dir-msg/libraries/appbase/application.cpp#L112-L148

**Comment by mvandeberg** [comment]
The message is now:

```
The default data directory is now '$HOME/.steemd' instead of '(pwd)/witness_node_data_dir'.
Please move you data directory to '$HOME/.steemd' or specificy '--data-dir=(pwd)/witness_node_data_dir' to continue using the current data directory.
```

`$HOME` and `(pwd)` are properly replaced in the actual error message.

**Comment by roadscape** [comment]
Looks great, one typo - "you" should be "your"

**Comment by sneak** [comment]
Excellent. That should cut down on complaints related to this change.

**Pull request merged**